src/node_buffer.cc

```diff
@@ -599,10 +599,6 @@ void Fill(const FunctionCallbackInfo<Value>& args) {
   size_t str_length;
   enum encoding enc;
 
-  // OOB Check. Throw the error in JS.
-  if (start > end || fill_length + start > ts_obj_length)
-    return args.GetReturnValue().Set(-2);
```

Comment:
This should stay as a CHECK.

Reply:
What's the reasoning?

Reply:
Changes (bugs) to the JS code should get caught at the C++ level, not allow buffer overruns.